File: kubik/kubik-server/src/main/java/com/cspinformatique/kubik/server/domain/product/service/SupplierService.java
package com.cspinformatique.kubik.server.domain.product.service;

import com.cspinformatique.kubik.server.model.product.Supplier;

public interface SupplierService {
	
	Iterable<Supplier> findAll();
	
	Supplier findByEan13(String ean13);
	
	Supplier findOne(Integer id);
	
	Supplier generateSupplierIfNotFound(String ean13);
	
	Supplier save(Supplier supplier);
}

File: kubik/kubik-server/src/main/webapp/public/js/product/supplier/supplier-service.js
(function(){
	var SUPPLIER_URL = "/supplier";
	
	angular
		.module("Kubik")
		.factory("supplierService", SupplierService);
	
	angular
		.module("KubikProductVehicule")
		.factory("supplierService", SupplierService);
	
	function SupplierService($http){
		return {
			findAll : findAll,
			findByEan13 : findByEan13
		}
		
		function findAll(){
			return $http
			    .get(SUPPLIER_URL)
			    .then(getSuccess);
			
			function getSuccess(response){
				return response.data;
			}
		}
		
		function findByEan13(ean13){
			return $http
				.get(SUPPLIER_URL + "?ean13=" + ean13)
				.then(getSuccess);
			
			function getSuccess(response){
				return response.data;
			}
		}
	}
})();